### PR TITLE
update docs with AMD, tracked resources, lifecycle methods

### PR DESCRIPTION
**Pull request by mjbradford89**
These changes update the docs on sennajs.com with AMD, tracked resources, lifecycle methods.

There are no changes to the API docs included.

### CODE REVIEW

**Comment by zenorocha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: src/public/docs/index.soy [hunk 1]
```diff
@@ -88,6 +88,18 @@
...
           </p>
         </p>
+        <p>
+          Senna.js is small, unobtrusive, and flexible enough so that it is easy to start a project with as well as integerate into existing applications.  Senna.js makes this easier by providing features like:
+          <p>
+            <strong>Options for loading:</strong> Senna.js can be included via AMD loading as well linked with a standard script tag and used via a global variable definiton.
+          </p>
+          <p>
+            <strong>Tracked resources:</strong> Track and manage page resources dynamically that are annoatated with the <span class="code">data-senna-track</span> html attribute.  Scripts and stylesheets can either be <span class="code">temporary</span> or <span class="code">permanent</span> resources.  They are then removed after navigation away from the page on which they are needed, or remain on the page permanently.
```
Comment (zenorocha) [inline src/public/docs/index.soy]: `annoatated ` > `annotated`

File: src/public/docs/index.soy [hunk 2]
```diff
@@ -383,6 +398,17 @@ FooScreen.prototype.getSurfaceContent = function(surfaceId) {
...
         {/literal}
+        <div class="title-header">
+          <h2 id="trackedresources">Tracked Resources</h2>
+        </div>
+
+        <p>
+          Senna.js provides an easy way to manage external style and script resources on the page.  When including resources in the <span class="code">&lt;head&gt;</span>, they can be annotated with the <span class="code">data-senna-track</span> attribute.  This attribute can be set to <span class="code">temporary</span> or <span class="code">permanent</span>, which determines whether a resource remains on the page permanently or is removed after navigation;
+        </p>
+        {literal}
+        <pre><code class="html">&lt;link data-senna-track="permanent" href="main.css" rel="stylesheet" type="text/css"&gt;
```
Comment (zenorocha) [inline src/public/docs/index.soy]: `type="text/css"` can be omitted

File: src/public/docs/index.soy [hunk 3]
```diff
@@ -88,6 +88,18 @@
             Not only that the list of features needed for a good SPA is huge, at the moment <a href="https://developer.mozilla.org/en-US/docs/Web/Guide/API/DOM/Manipulating_the_browser_history" target="blank">History API</a> presents many cross-browser inconsistencies that can get pretty complicated to be handled without a SPA engine. Therefore, if you decide to start from scratch your amazing MV* framework that supports SPA, think about using some engine as core, like Senna.js.
```
Comment (zenorocha) [inline src/public/docs/index.soy]: Remove `like Senna.js` since we repeat `Senna.js` in the next phrase.

File: src/public/docs/index.soy [hunk 4]
```diff
@@ -88,6 +88,18 @@
...
           </p>
         </p>
+        <p>
+          Senna.js is small, unobtrusive, and flexible enough so that it is easy to start a project with as well as integerate into existing applications.  Senna.js makes this easier by providing features like:
+          <p>
+            <strong>Options for loading:</strong> Senna.js can be included via AMD loading as well linked with a standard script tag and used via a global variable definiton.
```
Comment (zenorocha) [inline src/public/docs/index.soy]: `Options for loading` > `AMD support`

File: src/public/docs/index.soy (hunk 1 above)
Comment (zenorocha) [inline src/public/docs/index.soy]: `Tracked resources` > `Trackable resources`

**Comment by zenorocha** [comment]
Could you mentioned AMD support on Step 1?

**Comment by mjbradford89** [comment]
hey @zenorocha, I updated the docs based on your comments.  thanks!

**Comment by zenorocha** [comment]
Thanks @mjbradford89!

**Pull request merged**